scripts/build.sh

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+script_name="${BASH_SOURCE:-$0}"
+script_path=$(realpath "$script_name")
+scripts_dir_path=$(dirname "$script_path")
+
+# shellcheck disable=SC1091
+source "${scripts_dir_path}/lib/_functions.sh"
+# shellcheck disable=SC1091
+source "${scripts_dir_path}/lib/_logger.sh"
+
+function usage {
+    cat <<EOF
+This script performs the following tasks in this specific order:
+1. Generates a list of Operator manifests for a SonataFlow project
+2. Builds the workflow image using 
+3. Optionally deploys.
+    3.1 Pushes the image to a registry (specidied in the image path)
+    3.2 Applies the generated manifests in the current k8s namespace
+
+Usage: 
+    $script_name [flags]
+
+Flags:
+    -i|--image string                 The image path to use for the workflow (required).
+    -b|--builder-image string         Overrides the image to use for building the workflow image.
+    -h|--help                         Prints this help message.
+    -r|--runtime-image string         Overrides the image to use for running the workflow.
+    -w|--workflow-directory string    Path to the directory containing the workflow's files (the 'src' directory). Default: current directory.
+    -P|--no-persistence               Skips adding persistence configuration to the sonataflow CR.
+       --apply                        Applies the generated manifests in the current namespace.
+       --push                         Pushes the image to the registry after building
+
+Notes: 
```

Comment:
Suggestion: add the `kn workflow` version needed
Maybe the script could even download it from URL and use it locally (i.e: download it in the tmp) if the version is not the right one, but that's a long shot

Reply:
Although I agree it is a good idea, IMHO, all the tools needed: `kubectl` or `oc`, maybe `git`, `kn`, `kn-workflow` and so on... all those should be set up separately. Perhaps having a builder image deployed somewhere with all those tools would be enough.